### PR TITLE
A3IDES-577 obtain progress vars

### PR DESCRIPTION
**Pull request by michalrudolf**

### CODE REVIEW

File: src/wui/wui.c [hunk 1]
```diff
@@ -52,15 +52,18 @@ void StartWebServerTask(void const *argument) {
 
         if (wui.wui_marlin_vars) {
             marlin_client_loop();
+            wui.wui_marlin_vars = marlin_update_vars(MARLIN_VAR_MSK_WUI);
```
Comment (joshymjose) [inline src/wui/wui.c]: do this in a function say, transfer_var...()

File: src/wui/progress_data_wrapper.cpp [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#include "../../../lib/Marlin/Marlin/src/gcode/lcd/M73_PE.h"
```
Comment (joshymjose) [inline src/wui/progress_data_wrapper.cpp]: writing the path to header is not nice way, you have to "target link library" marlin library

File: src/wui/progress_data_wrapper.cpp [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+#include "progress_data_wrapper.h"
+#include <stdio.h>
+
+extern "C" {
+
+bool is_percentage_valid(uint32_t print_dur) {
+    return oProgressData.oPercentDone.mIsActual(print_dur);
```
Comment (joshymjose) [inline src/wui/progress_data_wrapper.cpp:8]: are these thread safe?

File: src/wui/wui_api.c [hunk 4]
```diff
@@ -41,17 +42,47 @@ const char *get_update_str(const char *header) {
     uint16_t flow_factor = (uint16_t)(web_vars_copy.flow_factor);
     const char *filament_material = filaments[get_filament()].name;
 
+    if (!web_vars_copy.sd_printing) {
+        return char_streamer("%s{"
+                             "\"temp_nozzle\":%d,"
```
Comment (joshymjose) [inline src/wui/wui_api.c:44]: this code needs refactoring, char_streamer has to be only at one place.

**Review by joshymjose — CHANGES_REQUESTED** [review]

File: src/wui/wui_api.c (hunk 4 above)
Comment (joshymjose) [reply src/wui/wui_api.c:44]: remove commented dead codes

File: src/wui/progress_data_wrapper.cpp (hunk 3 above)
Comment (michalrudolf) [reply src/wui/progress_data_wrapper.cpp:8]: Shouldn't be a problem. They are as safe as GUI's progress calculations.

**Review by joshymjose — APPROVED** [review]

**Pull request merged**